st2actions/st2actions/runners/actionchainrunner.py

```diff
@@ -494,10 +509,8 @@ def _run_action(self, liveaction, wait_for_completion=True):
             LOG.exception('Failed to schedule liveaction.')
             raise e
 
-        while (wait_for_completion and
-               liveaction.status != LIVEACTION_STATUS_SUCCEEDED and
-               liveaction.status != LIVEACTION_STATUS_FAILED):
-            eventlet.sleep(1)
+        while (wait_for_completion and liveaction.status not in COMPLETED_STATES):
```

Comment:
Previously, if `wait_for_completion` was `True` (it is by default and we never explicitly pass `False`) and if user canceled the action (task) while it was still running this while loop would never finish since termination condition would never be met (liveaction status would be set to canceled for which we didn't check).